contracts/staking/contracts/src/vaults/EthVault.sol

```diff
@@ -0,0 +1,110 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "../interfaces/IEthVault.sol";
+import "./MixinVaultCore.sol";
+
+
+/// @dev This vault manages ETH.
+contract EthVault is
+    Authorizable,
+    IEthVault,
+    IVaultCore,
+    MixinVaultCore
+{
+
+    using LibSafeMath for uint256;
+
+    // mapping from Owner to ETH balance
+    mapping (address => uint256) internal balances;
+
+    /// @dev Constructor.
+    // solhint-disable-next-line no-empty-blocks
+    constructor() public {}
+
+    /// @dev Deposit an `amount` of ETH from `owner` into the vault.
+    /// Note that only the Staking contract can call this.
+    /// Note that this can only be called when *not* in Catostrophic Failure mode.
+    /// @param owner of ETH Tokens.
+    function depositFor(address owner)
+        external
+        payable
+    {
+        // update balance
+        uint256 amount = msg.value;
+        balances[owner] = balances[owner].safeAdd(msg.value);
+
+        // notify
+        emit EthDepositedIntoVault(msg.sender, owner, amount);
+    }
+
+    /// @dev Withdraw an `amount` of ETH to `msg.sender` from the vault.
+    /// Note that only the Staking contract can call this.
+    /// Note that this can only be called when *not* in Catostrophic Failure mode.
+    /// @param amount of ETH to withdraw.
+    function withdraw(uint256 amount)
```

Comment:
A lot of these functions are exactly the same as in the ZrxVault. Maybe the repeat functions should live in `MixinVaultCore`?